### ISSUE TITLE
SSR does not handle multiple text nodes well

### ISSUE THREAD

**Issue by mikesol** [mikesol/purescript-deku#29]
It seems to drop nodes if there are several in the same element.

**Comment by mikesol** [comment]
It looks like one way to handle this in the SSR engine is for the text node to write its ID as a data attribute to its parent node in the SSR, and the value of this attribute can be which child it is. Then, we can use the proper index when hydrating. Currently, the index is hardcoded at 0:

```javascript
	if (
		tryHydration &&
		// hack
		a.parent.value0 &&
		(dom = document.body
			// hack
			.querySelectorAll("[data-deku-ssr-" + a.parent.value0 + "]")
			.item(0))
	) {
		state.units[ptr] = {
			// if we've done ssr for a text node, it will be a span,
			// so we want to get the child node
			main: dom.childNodes[0],
			parent: a.parent,
			scope: a.scope,
		};
	}
```

**Comment by mikesol** [comment]
Fixed in `0.4.10`.
